SmartCarMQTTApp/app/src/androidTest/java/com/example/smartcarmqttapp/MqttCarTest.java

```diff
@@ -0,0 +1,76 @@
+package com.example.smartcarmqttapp;
+
+import static org.junit.Assert.*;
+import static com.example.smartcarmqttapp.IntegrationTestUtils.*;
+
+import android.content.Context;
+
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+import androidx.test.platform.app.InstrumentationRegistry;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+
+/**
+ * Tests for checking if the communication with the car works.
+ */
+@RunWith(AndroidJUnit4.class)
+public class MqttCarTest {
+    /**
+     * Fault tolerance (absolute values)
+     */
+    public static final class FaultTolerance {
+        public static final double Speed = 0.01;
+    }
+
+    private static Context appContext;
+    private static MqttCar car;
+
+    /**
+     * Tries to setup the connection to a mqtt car.
+     *
+     * @throws Exception when the car setup fails
+     */
+    @BeforeClass
+    public static void setupCar() throws Exception {
+        appContext = InstrumentationRegistry.getInstrumentation().getTargetContext();
+        car = new MqttCar(appContext, () -> {});
+
+        if (!eventually(() -> car.isConnected())) {
+            throw new Exception("Failed to connect to the car!");
+        }
+    }
+
+    // ToDo: Add tests for the mqtt car here
+
+    @Test
+    public void GivenAStandingCar_WhenSpeedingFully_ThenTheSpeedShouldMatch() throws Exception {
```

Comment:
Any reason to have a method name this long? A simple comment should suffice @Lancear

Reply:
The idea behind this name is that when running the test the runner displays the method names for each test. Then you immediatly know which scenario went wrong and dont have to go to the test first to read a comment.